plugins/intercom/stores/IntercomStore.js

```diff
@@ -0,0 +1,99 @@
+import PluginSDK from "PluginSDK";
+
+import GetSetBaseStore from "#SRC/js/stores/GetSetBaseStore";
+import {
+  DCOS_METADATA_CHANGE,
+  METADATA_CHANGE
+} from "#SRC/js/constants/EventTypes";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+
+import { INTERCOM_CHANGE } from "../constants/EventTypes";
+
+class IntercomStore extends GetSetBaseStore {
+  constructor() {
+    super(...arguments);
+
+    this.isListening = false;
+
+    this.onDCOSMetadataChange = this.onDCOSMetadataChange.bind(this);
+    this.onMetadataChange = this.onMetadataChange.bind(this);
+
+    PluginSDK.addStoreConfig({
+      store: this,
+      storeID: this.storeID,
+      events: {
+        intercomChange: INTERCOM_CHANGE
+      },
+      unmountWhen() {
+        return true;
+      },
+      listenAlways: true
+    });
+  }
+
+  removeChangeListener(eventName, callback) {
+    this.removeListener(eventName, callback);
+
+    if (this.isListening) {
+      this.isListening = false;
+
+      MetadataStore.removeChangeListener(
+        DCOS_METADATA_CHANGE,
+        this.onDCOSMetadataChange
+      );
+
+      MetadataStore.removeChangeListener(
+        METADATA_CHANGE,
+        this.onMetadataChange
+      );
+    }
+  }
+
+  addChangeListener(eventName, callback) {
+    this.on(eventName, callback);
+
+    if (!this.isListening) {
```

Comment:
Related to https://github.com/dcos/dcos-ui/pull/3206/commits/734dec5d478cdd03a5bc1fb5b65750598ee3c0fd#r213110433